aep/general/0231/aep.md.j2

```diff
@@ -1,5 +1,182 @@
+    references.
+  - The parameter should define the pattern of the resource path values.
+  - The parameter should define the maximum number of paths allowed.
+- Other parameters besides `paths` **may** be "hoisted" from the [standard Get
+  request][get-request].
+- Batch Get **should not** support pagination because transactionality across
+  API calls would be extremely difficult to implement or enforce, and the
+  request defines the exact scope of the response anyway.
+- The request **must not** contain any other required parameters, and **should
+  not** contain other optional parameters except those described in this or
+  another AEP.
+
+{% tab proto -%}
+
+```proto
+rpc BatchGetBooks(BatchGetBooksRequest) returns (BatchGetBooksResponse) {
+  option (google.api.http) = {
+    get: "/v1/{parent=publishers/*}/books:batchGet"
+  };
+}
+
+message BatchGetBooksRequest {
+  // The parent resource shared by all books being retrieved.
+  // Format: publishers/{publisher}
+  // If this is set, the parent of all of the books specified in `paths`
+  // must match this field.
+  string parent = 1 [
+    (google.api.resource_reference) = {
+      child_type: "library.com/Book"
+    }];
+
+  // The paths of the books to retrieve.
+  // A maximum of 1000 books can be retrieved in a batch.
+  // Format: publishers/{publisher}/books/{book}
+  repeated string paths = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.com/Book"
+    }];
+}
+```
+
+- The request schema **must** match the method name, with `Request` suffix.
+- A `parent` field **should** be included, unless the resource being retrieved
+  is a top-level resource, to facilitate inclusion in the URI as well to permit
+  a single permissions check. If a caller sets this field, and the parent
+  collection in the path of any resource being retrieved does not match, the
+  request **must** fail.
+  - This field **should** be required if only 1 parent per request is allowed.
+  - The field **should** identify the [resource type][aep-122-parent] that it
+    references.
+  - The comment for the field **should** document the resource pattern.
+- There **must not** be a body key in the `google.api.http` annotation.
+
+{% tab oas %}
+
+{% sample 'batchget.oas.yaml', 'paths' %}
+
+- The `paths` parameter **must** be a query parameter which accepts an array of
+  resource paths specifying the resources to retrieve. The parameter **should**
+  be named `paths`.
+  - If no resource paths are provided, the API **should** error with status
+    code `400 Bad Request`.
+  - If the collection in the path of any resource does not match the collection
+    of the request URL, the request **must** fail.
+- The method definition **must not** have a `requestBody` defined.
+
+{% endtabs %}
+
+### Response
+
+The response for a batch get method **should** be specified with the following
```

Comment:
```suggestion
The response for a batch get method **must** be specified with the following
```

Feels weird to have a should that includes multiple musts. Why not enforce the pattern?